src/role-access/containers/allocate-role/allocate-role-home/allocate-role-home.component.ts

```diff
@@ -0,0 +1,141 @@
+import { AllocateRoleState } from '../../../models';
+import { AllocateRoleNavigationEvent } from '../../../models/allocate-role-navigation-event.enum';
+import { AllocateRoleNavigation } from '../../../models/allocate-role-navigation.interface';
+import * as fromFeature from '../../../store';
+import { AllocateRoleCheckAnswersComponent } from '../allocate-role-check-answers/allocate-role-check-answers.component';
+import { AllocateRoleSearchPersonComponent } from '../allocate-role-search-person/allocate-role-search-person.component';
+import { ChooseAllocateToComponent } from '../choose-allocate-to/choose-allocate-to.component';
+import { ChooseDurationComponent } from '../choose-duration/choose-duration.component';
+import { ChooseRoleComponent } from '../choose-role/choose-role.component';
+
+@Component({
+  selector: 'exui-allocate-role-home',
+  templateUrl: './allocate-role-home.component.html',
+})
+export class AllocateRoleHomeComponent implements OnInit, OnDestroy {
+
+  @ViewChild('chooseRole', {read: ChooseRoleComponent})
+  public chooseRoleComponent: ChooseRoleComponent;
+
+  @ViewChild('chooseAllocateTo', {read: ChooseAllocateToComponent})
+  public chooseAllocateToComponent: ChooseAllocateToComponent;
+
+  @ViewChild('searchPerson', {read: AllocateRoleSearchPersonComponent})
+  public searchPersonComponent: AllocateRoleSearchPersonComponent;
+
+  @ViewChild('chooseDuration', {read: ChooseDurationComponent})
+  public chooseDurationComponent: ChooseDurationComponent;
+
+  @ViewChild('checkAnswers', {read: AllocateRoleCheckAnswersComponent})
+  public checkAnswersComponent: AllocateRoleCheckAnswersComponent;
+
+  public chooseRoleVisibilityStates = chooseRoleVisibilityStates;
+  public chooseAllocateToVisibilityStates = chooseAllocateToVisibilityStates;
+  public searchPersonVisibilityStates = searchPersonVisibilityStates;
+  public chooseDurationVisibilityStates = chooseDurationVisibilityStates;
+  public checkAnswersVisibilityStates = checkAnswersVisibilityStates;
+
+  public navEvent: AllocateRoleNavigation;
+
+  public allocateRoleStateDataSub: Subscription;
+
+  public navigationCurrentState: AllocateRoleState;
+  public caseId: string;
+
+  constructor(private readonly store: Store<fromFeature.State>,
+              private readonly router: Router) { }
+
+  public ngOnInit(): void {
+    this.allocateRoleStateDataSub = this.store.pipe(select(fromFeature.getAllocateRoleState)).subscribe(
+      allocateRoleStateData => {
+        this.navigationCurrentState = allocateRoleStateData.state;
+        this.caseId = allocateRoleStateData.caseId;
+      }
+    );
+  }
+
+  public isComponentVisible(currentNavigationState: AllocateRoleState, requiredNavigationState: AllocateRoleState[]): boolean {
+    return requiredNavigationState.includes(currentNavigationState);
+  }
+
+  public onNavEvent(event: AllocateRoleNavigationEvent): void {
+    this.navEvent = {
+      event,
+      timestamp: Date.now()
```

Comment:
Minor query - Why is the timestamp necessary? Is it going to be used?

Reply:
it's not used in the business requirement, but technically this will tell when the user trigger the event.

Reply:
Ok sure - I suppose we could use it in the future